public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -674,6 +683,41 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
       this._scrollRef?.scrollTo(0, this._prevScrollPos!);
     }
   }
+
+  getSaveModel(): Dashboard | DashboardV2Spec {
+    if (isV2Dashboard(this._initialSaveModel!)) {
+      return transformSceneToSaveModelSchemaV2(this) as DashboardV2Spec;
+    }
+    return transformSceneToSaveModel(this);
+  }
+
+  getSaveAsModel(options: SaveDashboardAsOptions): Dashboard {
+    const saveModel = this.getSaveModel();
+    if (isV2Dashboard(saveModel)) {
+      throw new Error('Save as is not supported for v2 dashboards');
+    } else {
+      return {
+        ...saveModel,
+        id: null,
+        uid: '',
+        title: options.title || '',
+        description: options.description || '',
+        tags: options.isNew || options.copyTags ? saveModel.tags : [],
+      };
+    }
+  }
+
+  getDashboardChanges(saveTimeRange?: boolean, saveVariables?: boolean, saveRefresh?: boolean): DashboardChangeInfo {
+    const changeInfo = getDashboardChangesFromScene(this, saveTimeRange, saveVariables, saveRefresh);
+
+    const hasFolderChanges = this.getInitialState()?.meta.folderUid !== this.state.meta.folderUid;
+
+    return {
+      ...changeInfo,
+      hasFolderChanges,
+      hasChanges: changeInfo.hasChanges || hasFolderChanges,
+    };
```

Comment:
think having these functions on a separate class that has 2 implementations would gett rid of all the special if statements in getDashboardChangesFromScene

Reply:
yeah, they did. 2nd pass is now pushed